### PR TITLE
Fix auto-correct for BlockAlignment

### PR DESCRIPTION
**Pull request by lumeet**
This is a minimalistic attempt to solve #1573. @jonas054, it would be great if you could provide some code failing examples in case this patch doesn't fulfill the criteria well enough.

### CODE REVIEW

File: lib/rubocop/cop/lint/block_alignment.rb [hunk 1]
```diff
@@ -148,12 +148,18 @@ def block_is_on_next_line?(begin_node, block_node)
...
 
+        def ancestor_on_same_line(node)
+          node.ancestors.reverse.find do |ancestor|
+            ancestor.loc.line == node.loc.line
+          end
+        end
+
         def autocorrect(node)
-          key = node.children.first
+          ancestor_node = ancestor_on_same_line(node)
```
Comment (jonas054) [inline lib/rubocop/cop/lint/block_alignment.rb]: Make calls downwards in the file when possible.

**Comment by jonas054** [comment]
Nice solution! I think there's a bug left in the code. It was there before, and I've opened a separate issue for it, #1579.

File: lib/rubocop/cop/lint/block_alignment.rb (hunk 1 above)
Comment (lumeet) [inline lib/rubocop/cop/lint/block_alignment.rb]: Can you elaborate a little? I don't really get that one.

Comment (jonas054) [inline lib/rubocop/cop/lint/block_alignment.rb]: I'd be happy to! :smile: Code should read like a newspaper article, meaning that you get the broad picture of what's going on at the top of the file, and the further down you get the more detailed it gets. So when you call `ancestor_on_same_line`, the definition of it should come after the call (so after `autocorrect`).

Comment (lumeet) [inline lib/rubocop/cop/lint/block_alignment.rb]: Aaah.. that's a good point, thanks. I started to interpret your comment totally wrong :smile:. I'll fix that along with a new guard clause.

**Comment by bquorning** [comment]
> Make calls downwards in the file when possible.

Is there a Cop for that? :-)

**Comment by bbatsov** [comment]
:+1:

**Pull request merged**